src/test/java/com/google/api/codegen/transformer/OutputTransformerTest.java

```diff
@@ -0,0 +1,79 @@
+/* Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.transformer;
+
+import static com.google.api.codegen.transformer.OutputTransformer.ScopeTable;
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.codegen.config.ProtoTypeRef;
+import com.google.api.codegen.config.TypeModel;
+import com.google.api.tools.framework.model.TypeRef;
+import org.junit.Before;
+import org.junit.Test;
+
+public class OutputTransformerTest {
+
+  private ScopeTable outer;
+  private ScopeTable inner;
+
+  @Before
+  public void setUp() {
+    outer = new ScopeTable();
+    inner = new ScopeTable(outer);
```

Comment:
I just realized, I think you have this swapped right? Let's have
```java
parent = new ScopeTable();
child = new ScopeTable(parent);
```
since that's the name used in the scope table.